pkg/controller/status/status.go

```diff
@@ -153,6 +153,14 @@ func (m *statusManager) OnCRNotFound() {
 	m.clearProgressing()
 	m.lock.Lock()
 	defer m.lock.Unlock()
+	// Remove status displayed for CR, if CR was available earlier but not now.
+	if m.enabled {
```

Comment:
I don't think we want to remove this here, all interactions with the API are done in the Run() loop. This function should only set the flag that the tigera-status should be removed, and then some part of the go routine in the Run() loop should remove the tigera-status (look here https://github.com/tigera/operator/blob/1ff13fc9f3d7962c591282f9442b6dd0e881c3a4/pkg/controller/status/status.go#L105).

 I would also consider if it makes sense to clear everything from the status manager (i.e. deployments, daemonsets,...).

Reply:
I've moved the logic to Run() loop and tested :+1: 

> I would also consider if it makes sense to clear everything from the status manager (i.e. deployments, daemonsets,...).
This is already done when m.enabled flag is set to false in OnCRNotFound()